indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -335,20 +343,81 @@ def _process(self):
+        RHRegistrationsActionBase._checkProtection(self)
+
+        # Check that template belongs to this event or a category that
+        # is a parent
+        if self.template.owner != self.event_new and self.template.owner.id not in self.event_new.category_chain:
+            raise Forbidden
+
+    def _process(self):
+        self.commit = False
+        cache = GenericCache('badge-printing')
+
+        config_params = cache.get(request.view_args['uuid'])
+        if not config_params:
+            raise NotFound
+
+        registration_ids = config_params.pop('registration_ids')
+        pdf = RegistrantsListToBadgesPDF(self.template, config_params, self.event_new, registration_ids)
+        return send_file('Badges-{}.pdf'.format(self.event_new.id), pdf.get_pdf(), 'PDF')
+
+
+class RHRegistrationsConfigBadges(RHRegistrationsActionBase):
     """Print badges for the selected registrations"""
 
+    def _checkParams(self, params):
+        # TODO: Any POST-based alternative to this?
+        RHManageRegFormBase._checkParams(self, params)
+        ids = set(request.args.getlist('registration_id'))
+        self.registrations = (Registration.query.with_parent(self.regform)
+                              .filter(Registration.id.in_(ids),
+                                      ~Registration.is_deleted)
+                              .order_by(*Registration.order_by_name)
+                              .all())
+        self.template_id = request.args.get('template_id')
+
     def _process(self):
-        badge_templates = self._conf.getBadgeTemplateManager().getTemplates().items()
-        badge_templates.sort(key=lambda x: x[1].getName())
+        badge_templates = sorted((tpl for tpl in get_all_templates(self.event_new) if tpl.type.name == 'badge'),
+                                 key=attrgetter('title'))
         pdf_options = WConfModifBadgePDFOptions(self._conf).getHTML()
-        badge_design_url = url_for('event_mgmt.confModifTools-badgePrinting', self.event_new)
-        create_pdf_url = url_for('event_mgmt.confModifTools-badgePrintingPDF', self.event_new)
-
-        return WPManageRegistration.render_template('management/print_badges.html', self._conf, regform=self.regform,
-                                                    templates=badge_templates, pdf_options=pdf_options,
-                                                    registrations=self.registrations,
-                                                    registration_ids=[x.id for x in self.registrations],
-                                                    badge_design_url=badge_design_url, create_pdf_url=create_pdf_url)
+        settings = event_badge_settings.get_all(self.event_new.id)
+        form = BadgeSettingsForm(self.event_new, **settings)
+        cache = GenericCache('badge-printing')
+        registrations = self.registrations or self.regform.registrations
+
+        if form.validate_on_submit():
+            data = dict(form.data)
```

Comment:
just `data = form.data` would be enough (even though it's a bit implicit... no strong opinion on whether the explicit copy is better or not in the end)